lib/source-loader/src/server/abstract-syntax-tree/traverse-helpers.js

```diff
@@ -20,10 +20,80 @@ export function splitSTORYOF(ast, source) {
 
   return parts;
 }
+
+function isFunctionVariable(declarations, includeExclude) {
+  return (
+    declarations &&
+    declarations.length === 1 &&
+    declarations[0].type === 'VariableDeclarator' &&
+    declarations[0].id &&
+    declarations[0].id.name &&
+    declarations[0].init &&
+    ['CallExpression', 'ArrowFunctionExpression', 'FunctionExpression'].includes(
+      declarations[0].init.type
+    ) &&
+    isExportStory(declarations[0].id.name, includeExclude)
+  );
+}
+
+function isFunctionDeclaration(declaration, includeExclude) {
+  return (
+    declaration.type === 'FunctionDeclaration' &&
+    declaration.id &&
+    declaration.id.name &&
+    isExportStory(declaration.id.name, includeExclude)
+  );
+}
+
+function getDescriptor(metaDeclaration, propertyName) {
+  const property =
+    metaDeclaration &&
+    metaDeclaration.declaration &&
+    metaDeclaration.declaration.properties.find(p => p.key && p.key.name === propertyName);
+  if (!property) {
+    return undefined;
+  }
+
+  const { type } = property.value;
+
+  switch (type) {
+    case 'ArrayExpression':
+      return property.value.elements.map(t => t.value);
```

Comment:
What's allowed in an Array? strings only?

Reply:
Updated